astropy/units/tests/test_quantity_decorator.py

```diff
@@ -323,16 +324,25 @@ def myfunc_args(x: "kilograam"):
 
 
 def test_type_annotations():
+    # Basic Annotation
     @u.quantity_input
     def myfunc_args(x: u.m, y: str):
         return x, y
 
-    in_quantity = 2 * u.m
-    in_string = "cool string"
+    i_q, i_str = 2 * u.m, "cool string"
+    o_q, o_str = myfunc_args(i_q, i_str)
+    assert o_q == i_q
+    assert o_str == i_str
 
-    quantity, string = myfunc_args(in_quantity, in_string)
-    assert quantity == in_quantity
-    assert string == in_string
+    # unit-aware Quantity Annotation
+    @u.quantity_input
+    def myfunc_args(x: Annotated[u.Quantity, u.m], y: str):
```

Comment:
I'm confused: is there no test here of just `x: u.Quantity[u.m]`?

Reply:
Adding it now.